client/src/lib/solana/sol-data.ts

```diff
@@ -189,42 +248,52 @@ export class SolData extends Assignable {
     return VERSION;
   }
 
-  static parseDIDReferenceArray(
+  static async parseDIDReferenceArray(
     dids: (string | DIDVerificationMethod)[] | undefined
-  ): string[] {
-    return DecentralizedIdentifier.parseMaybeArray(dids).map(
+  ): Promise<(string | undefined)[]> {
+    return (await DecentralizedIdentifier.parseMaybeArray(dids)).map(
```

Comment:
nested asyncs are a little hard to reason about. prefer either splitting into a separate line or using promises i.e. then() instead. (But then(....map()) is not so nice and TS has no native flatmap function that recognises promises.)

Reply:
I'm not sure the .then is much better but I added it for comparison. Can always go back